### PR TITLE
feat: move messages to junk folder

### PR DESCRIPTION
**Pull request by kesselb**
For https://github.com/nextcloud/mail/issues/4101

**To Do**

- [x] Decide: Merge JunkSettings into AccountDefaultSettings
- [x] ~Decide: Rename "Mark as Spam" to "Move to Spam" with checkbox "Move message to Junk" enabled~
- [x] Inline documentation about the action/getter behavior
- [x] Testing

**Test cases**

| Message List | Message View |
| --- | --- |
|Screenshot from 2023-07-07 18-45-40 |Screenshot from 2023-07-07 18-46-01|

1. Add a new account. See a different icon for an existing folder/mailbox with the special role "junk".
2. Open account settings. A section "Junk settings" exist. The checkbox "Move message to Junk" is off by default.
3. Open action menu from the message list. Click "Mark as Spam". The message is flagged as spam, but not moved. 
4. Open action menu from the message view. Click "Mark as Spam". The message is flagged as spam, but not moved. 
5. Open account settings. Toggle checkbox "Move message to Junk".
6. Open action menu from the message list. Click "Mark as Spam". The message is flagged as spam and moved.
7. Open action menu from the message view. Click "Mark as Spam". The message is flagged as spam and moved. 
8. Open junk mailbox and repeat 6 and 7.

### CODE REVIEW

File: lib/Listener/MoveJunkListener.php [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+				]);
+			}
+		} elseif (!$event->isSet() && 'INBOX' !== $mailbox->getName()) {
+			try {
+				$this->mailManager->moveMessage(
+					$account,
+					$mailbox->getName(),
+					$event->getUid(),
+					$account,
+					'INBOX',
```
Comment (ChristophWurst) [inline lib/Listener/MoveJunkListener.php:92]: can we assume the name to be always exactly like that? what about capitalization? personal namspace?

**Review by jancborchardt — COMMENTED** [review]
> Decide: Rename "Mark as Spam" to "Move to Spam" with checkbox "Move message to Junk" enabled

It should stay a simple "Mark as spam" action. It is expected from (the majority of) other clients that "Mark as spam" will also move it to spam/junk.

**Comment by kesselb** [comment]
image

I will add the junk mailbox to default folders.

**Review by ChristophWurst — APPROVED** [review]
Works

**Comment by ChristophWurst** [comment]
```
OCA\Mail\Tests\Integration\Db\MailAccountTest::testMailAccountConstruct
Failed asserting that two arrays are equal.
--- Expected
+++ Actual
@@ @@
     'quotaPercentage' => null
     'trashRetentionDays' => 60
     'name' => 'Peter Parker'
+    'junkMailboxId' => null
+    'moveJunk' => false
 )
```

**Comment by jancborchardt** [comment]
Follow-up as talked with @ChristophWurst and mentioned above:
>  It is expected from (the majority of) other clients that "Mark as spam" will also move it to spam/junk.

Meaning the toggle to need to enable this functionality can be removed as it should be enabled by default.

**Pull request merged**